src/render.rs

```diff
@@ -16,29 +16,68 @@ pub(crate) fn render_group<F: FnMut(&mut Scene, &usvg::Node)>(
+                    usvg::BlendMode::Overlay => vello::peniko::Mix::Overlay,
+                    usvg::BlendMode::Darken => vello::peniko::Mix::Darken,
+                    usvg::BlendMode::Lighten => vello::peniko::Mix::Lighten,
+                    usvg::BlendMode::ColorDodge => vello::peniko::Mix::ColorDodge,
+                    usvg::BlendMode::ColorBurn => vello::peniko::Mix::ColorBurn,
+                    usvg::BlendMode::HardLight => vello::peniko::Mix::HardLight,
+                    usvg::BlendMode::SoftLight => vello::peniko::Mix::SoftLight,
+                    usvg::BlendMode::Difference => vello::peniko::Mix::Difference,
+                    usvg::BlendMode::Exclusion => vello::peniko::Mix::Exclusion,
+                    usvg::BlendMode::Hue => vello::peniko::Mix::Hue,
+                    usvg::BlendMode::Saturation => vello::peniko::Mix::Saturation,
+                    usvg::BlendMode::Color => vello::peniko::Mix::Color,
+                    usvg::BlendMode::Luminosity => vello::peniko::Mix::Luminosity,
+                };
+
+                match g
+                    .clip_path()
+                    // support clip-path with a single path
+                    .and_then(|path| path.root().children().first())
+                {
+                    Some(usvg::Node::Path(clip_path)) => {
                         let local_path = util::to_bez_path(clip_path);
                         scene.push_layer(
                             BlendMode {
-                                mix: vello::peniko::Mix::Clip,
+                                mix,
                                 compose: vello::peniko::Compose::SrcOver,
                             },
-                            1.0,
+                            alpha,
                             transform,
                             &local_path,
                         );
-                        pushed_clip = true;
+                    }
+                    _ => {
+                        // Use bounding box as the clip path.
+                        let bounding_box = g.layer_bounding_box();
+                        let rect = vello::kurbo::Rect::from_origin_size(
+                            (bounding_box.x(), bounding_box.y()),
+                            (bounding_box.width() as f64, bounding_box.height() as f64),
+                        );
+                        scene.push_layer(
+                            BlendMode {
+                                mix,
+                                compose: vello::peniko::Compose::SrcOver,
+                            },
+                            alpha,
+                            transform,
+                            &rect,
+                        );
                     }
                 }
 
                 render_group(scene, g, Affine::IDENTITY, error_handler);
 
-                if pushed_clip {
-                    scene.pop_layer();
-                }
+                scene.pop_layer();
```

Comment:
If I'm not mistaken, we only popped a layer if we had to insert a clip layer here. Given that normal layers or other blend modes can be used now, should we always be popping?

Have you tested this with several images to ensure this didn't cause an inadvertent behavior?

Reply:
I have tested with several svgs, it works normally. The reason it's always popping is because we're always pushing a layer to cater for group transparency, we could optimize this a little by skipping a layer if `BlendMode == BlendMode::Normal` and `opacity == 1.0`

Reply:
Seems fine to me, as long as it works.

Reply:
I'm surprised that this still works; in current Vello, clip layers are:
1) Somewhat expensive to evaluate
2) Likely to fail very quickly once you get more than four layers deep.

I think the optimisation path is very important to have here